theano/tensor/opt.py

```diff
@@ -234,6 +237,11 @@ def inplace_elemwise_optimizer(fgraph):
         else:
             update_outs = []
 
+        protected_inputs = [
+            f.protected for f in fgraph._features if
+            isinstance(f, theano.compile.function_module.Supervisor)]
+        protected_inputs = sum(protected_inputs, [])  # flatten the list
+        protected_inputs.extend(fgraph.outputs)
```

Comment:
This is just code moved outside the loop as it is constant during all the loop iteration.